### PR TITLE
Improvements and shared update

### PR DESCRIPTION
**Pull request by kaulex99**

### CODE REVIEW

File: WaiterRobot/Core/Mvi/ObservableViewModel.swift [hunk 1]
```diff
@@ -6,29 +6,26 @@
...
 
-        activate()
-    }
-
-    deinit {
-        actual.onCleared()
-        task?.cancel()
+        Task {
+            await activate()
+        }
```
Comment (dev-Fabi) [inline WaiterRobot/Core/Mvi/ObservableViewModel.swift]: Only if `subscribe == true`

File: WaiterRobot/LaunchScreen.swift [hunk 2]
```diff
@@ -69,3 +70,31 @@ struct LaunchScreen: View {
...
+                    }
+                    .navigationTitle("Test")
+                    .navigationBarTitleDisplayMode(.inline)
+                    .padding()
+            }
+        } else {
+            Text("Test")
+        }
+    }
+}
```
Comment (dev-Fabi) [inline WaiterRobot/LaunchScreen.swift:100]: Waht's this?

File: WaiterRobot/Ui/Login/LoginScannerScreen.swift [hunk 3]
```diff
@@ -8,7 +8,7 @@ struct LoginScannerScreen: View {
 
     @StateObject private var viewModel = ObservableLoginScannerViewModel()
 
-    private let simulatedData = "https://lava.kellner.team/ml/signIn?token=w7wF6pgYA6Ssm3VBH-rSFL6if70&purpose=SIGN_IN"
+    private let simulatedData = "https://lava.kellner.team/ml/signIn?purpose=SIGN_IN&token=xWEP33DuYhJzUvQ6clywKPCM_Oa5NymihpJk4-_EGHV3D_f10YSKL_2hOYV3"
```
Comment (dev-Fabi) [inline WaiterRobot/Ui/Login/LoginScannerScreen.swift:11]: We should not commit valid tokens ;)

File: WaiterRobot/Ui/Login/RegisterScreen.swift [hunk 4]
```diff
@@ -50,7 +51,13 @@ struct RegisterScreen: View {
                 Spacer()
 
                 Button {
-                    viewModel.actual.onRegister(name: name, createToken: createToken)
+                    viewModel.actual.onRegister(
+                        name: name,
+                        registerLink: DeepLink.AuthRegisterLink(
+                            token: deepLink.token,
+                            apiBase: deepLink.apiBase // TODO: use correct id
+                        )
```
Comment (dev-Fabi) [inline WaiterRobot/Ui/Login/RegisterScreen.swift:59]:
```suggestion
                        registerLink: deepLink
```

File: WaiterRobot/Ui/Order/OrderListItem.swift [hunk 5]
```diff
@@ -17,63 +17,51 @@ struct OrderListItem: View {
...
-                }
-                .padding()
-                .navigationTitle(localize.order.addNoteDialog.title(value0: name))
-                .navigationBarTitleDisplayMode(.inline)
+            if #available(iOS 16.0, *) {
+                orderProductNote()
+//                    .presentationDetents([.medium, .large])
+            } else {
+                orderProductNote()
             }
```
Comment (dev-Fabi) [inline WaiterRobot/Ui/Order/OrderListItem.swift:65]: For both branches the same?

File: WaiterRobot/Ui/Order/OrderListItem.swift [hunk 6]
```diff
@@ -104,27 +92,33 @@ struct OrderListItem: View {
             .tint(.green)
         })
     }
+
+    private func orderProductNote() -> some View {
+        OrderProductNoteView(name: "test", noteText: $editedNote, onSaveNote: onSaveNote)
```
Comment (dev-Fabi) [inline WaiterRobot/Ui/Order/OrderListItem.swift:97]:
```suggestion
        OrderProductNoteView(name: name, noteText: $editedNote, onSaveNote: onSaveNote)
```

File: WaiterRobot/WaiterRobotApp.swift [hunk 7]
```diff
@@ -24,7 +23,8 @@ struct WaiterRobotApp: App {
             appBuild: Int32(readFromInfoPlist(withKey: "CFBundleVersion"))!,
             phoneModel: UIDevice.current.model,
             os: OS.Ios(version: UIDevice.current.systemVersion),
-            apiBaseUrl: readFromInfoPlist(withKey: "API_BASE")
+            allowedHostsCsv: "*", // TODO: separate according to selected target (comma separated list)
```
Comment (dev-Fabi) [inline WaiterRobot/WaiterRobotApp.swift:26]: TODO :)

**Review by dev-Fabi — COMMENTED** [review]
Nice :)

File: WaiterRobot/Ui/Login/LoginScannerScreen.swift (hunk 3 above)
Comment (kaulex99) [reply WaiterRobot/Ui/Login/LoginScannerScreen.swift:11]: haha already invalid xD

File: project.yml [hunk 8]
```diff
@@ -130,8 +130,8 @@ targets:
       identifier: "org.datepollsystems.waiterrobot.beta"
       displayName: "lava.kellner.team"
       deeplinkDomain: "lava.kellner.team"
-      apiBase: "https://lava.kellner.team/api"
       versionSuffix: "lava"
+      allowedHosts: "my.kellner.team,lava.kellner.team"
```
Comment (dev-Fabi) [inline project.yml:134]: Fine but on android I use "*" so that the lava version can also be tested with a local backend :)

**Review by dev-Fabi — APPROVED** [review]

**Pull request merged**